src/melee.cpp

```diff
@@ -248,6 +248,25 @@ void player::roll_all_damage( bool crit, damage_instance &di, bool average, cons
     roll_stab_damage( crit, di, average, weap );
 }
 
+static void melee_train( player &p, int lo, int hi ) {
+    p.practice( skill_melee, rng( lo, hi ) );
```

Comment:
Should probably train for 50%-75% of primary weapon skill. Otherwise it will always be higher than any weapon skill.